frontend/src/metabase/admin/performance/types.tsx

```diff
@@ -0,0 +1,152 @@
+import { c, t } from "ttag";
+import type { AnySchema } from "yup";
+import type { SchemaObjectDescription } from "yup/lib/schema";
+
+import {
+  doNotCacheStrategyValidationSchema,
+  durationStrategyValidationSchema,
+  inheritStrategyValidationSchema,
+  strategyValidationSchema,
+  ttlStrategyValidationSchema,
+} from "./validation";
+
+type StrategyData = {
+  label: string;
+  shortLabel?: string;
+  validateWith: AnySchema;
+};
+
+export type StrategyType = "nocache" | "ttl" | "duration" | "inherit";
+
+/** Cache invalidation strategies and related metadata */
+export const Strategies: Record<StrategyType, StrategyData> = {
+  ttl: {
+    label: t`TTL: When the time-to-live (TTL) expires`,
+    shortLabel: c("'TTL' is short for 'time-to-live'").t`TTL`,
+    validateWith: ttlStrategyValidationSchema,
+  },
+  duration: {
+    label: t`Duration: after a specific number of hours`,
+    validateWith: durationStrategyValidationSchema,
+    shortLabel: t`Duration`,
+  },
+  nocache: {
+    label: t`Don't cache results`,
+    validateWith: doNotCacheStrategyValidationSchema,
+    shortLabel: t`No caching`,
+  },
+  inherit: {
+    label: t`Use default`,
+    validateWith: inheritStrategyValidationSchema,
+  },
+};
```

Comment:
this isn't a type, this should probably be in a `util` or `constants` file

Reply:
There are closely related and interdefined utils, constants, and types, so I've put most of them into a new `strategies.ts` file (containing strategy-related code). The Strategy and Config types now live in metabase-types/api, though.